src/Module/Profile/Photos/Upload.php

```diff
@@ -0,0 +1,277 @@
+use Friendica\Module\BaseApi;
+use Friendica\Module\Response;
+use Friendica\Navigation\SystemMessages;
+use Friendica\Network\HTTPException\InternalServerErrorException;
+use Friendica\Object\Image;
+use Friendica\Util\Images;
+use Friendica\Util\Profiler;
+use Friendica\Util\Strings;
+use Psr\Log\LoggerInterface;
+
+class Upload extends \Friendica\BaseModule
+{
+	/** @var Database */
+	private $database;
+
+	/** @var IHandleUserSessions */
+	private $userSession;
+
+	/** @var SystemMessages */
+	private $systemMessages;
+
+	/** @var IManageConfigValues */
+	private $config;
+
+	/** @var bool */
+	private $isJson = false;
+
+	public function __construct(IManageConfigValues $config, SystemMessages $systemMessages, IHandleUserSessions $userSession, Database $database, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->database       = $database;
+		$this->userSession    = $userSession;
+		$this->systemMessages = $systemMessages;
+		$this->config         = $config;
+	}
+
+	protected function post(array $request = [])
+	{
+		$this->isJson = !empty($request['response']) && $request['response'] == 'json';
+
+		$album = trim($request['album'] ?? '');
+
+		if (empty($_FILES['media'])) {
+			$user = $this->database->selectFirst('owner-view', ['id', 'uid', 'nickname', 'page-flags'], ['nickname' => $this->parameters['nickname'], 'blocked' => false]);
+		} else {
+			$user = $this->database->selectFirst('owner-view', ['id', 'uid', 'nickname', 'page-flags'], ['uid' => BaseApi::getCurrentUserID() ?: null, 'blocked' => false]);
+		}
+
+		if (!$this->database->isResult($user)) {
+			$this->logger->warning('User is not valid', ['nickname' => $this->parameters['nickname'], 'user' => $user]);
+			$this->exit(404, $this->l10n->t('User not found.'));
+		}
+
+		/*
+		 * Setup permissions structures
+		 */
+		$can_post   = false;
+		$visitor    = 0;
+		$contact_id = 0;
+
+		$page_owner_uid  = $user['uid'];
+		$default_cid     = $user['id'];
+
+		if ($this->userSession->getLocalUserId() && $this->userSession->getLocalUserId() == $page_owner_uid) {
+			$can_post = true;
+		} elseif ($user['page-flags'] == User::PAGE_FLAGS_COMMUNITY && !$this->userSession->getRemoteContactID($page_owner_uid)) {
+			$contact_id = $this->userSession->getRemoteContactID($page_owner_uid);
+			$can_post   = $this->database->exists('contact', ['blocked' => false, 'pending' => false, 'id' => $contact_id, 'uid' => $page_owner_uid]);
+			$visitor    = $contact_id;
+		}
+
+		if (!$can_post) {
+			$this->logger->warning('No permission to upload files', ['contact_id' => $contact_id, 'page_owner_uid' => $page_owner_uid]);
+			$this->exit(403, $this->l10n->t('Permission denied.'), true);
+		}
+
+		if (empty($_FILES['userfile']) && empty($_FILES['media'])) {
```

Comment:
There's a specific class for files, @annando has created under Utils, I'm currently on my mobile phone so I cannot exactly say which one it is :D

Reply:
I could only find `Util\HttpInput` but it looks like the files array are just added to the `$request` array provided to `post()`.